synapse/lib/view.py

```diff
@@ -777,6 +775,34 @@ async def setLayers(self, layers):
         await self.info.set('layers', layers)
         await self.core.feedBeholder('view:setlayers', {'iden': self.iden, 'layers': layers}, gates=[self.iden, layers[0]])
 
+        await self._calcChildViews()
+        self.core._calcViewsByLayer()
+
+    async def _calcChildViews(self):
+
+        todo = collections.deque([self])
+
+        byparent = collections.defaultdict(list)
+        for view in self.core.views.values():
+            if view.parent is None:
+                continue
+
+            byparent[view.parent].append(view)
+
+        while todo:
+
+            view = todo.pop()
+
+            for child in byparent.get(view, ()):
+
+                layers = list(view.layers)
+                layers.append(child.layers[-1])
```

Comment:
Isn't `child.layers[-1]` the first layer in the bottom view. Do you want the top layer of the view? That should be `child.layers[0]`, right?

Reply:
Check out the layer ordering in this screenshot. The first print is from `test_lib_stormtypes.py:3470` and the second is from `test_lib_stormtypes.py:3476`.
![image](https://github.com/vertexproject/synapse/assets/11776738/ad44612d-a568-4ab9-a2de-cd7642e546d4)

Notice the top and bottom layers are identical after adding the new layer.